### PR TITLE
Custom File name Feature

### PR DESCRIPTION
**Pull request by nishant23122000**
## Current behavior
Till now, we have only a dropdown to select the file type which user wants to download and when we click on the option provided on the dropdown then the file downloaded automatically and the name of the is always **generate**

## Expected behavior
Now as I mention in issue #780, we have a new feature to add custom filename. check below steps

step 1: Click on the download button on the top section, the dialog box will open
step 2:  in the dialog box enter the file name that you want 
step 3: Select the file type that you want to download like pdf or png
step 4: Click on the download button at the bottom of the dialog
step 5: you will see that file downloaded with you the custom filename

https://user-images.githubusercontent.com/53515714/112717690-5437e800-8f14-11eb-8829-103d0e4d6503.mp4

## Exception situation:
There is also error handling if the user download the file without providing the filename

PS:  You can see the current filetype which have a  *  appended

### CODE REVIEW

File: src/pages/Editor/sections/Settings/DownloadFileModal.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                }
+
+                <label className={styles.modalLabel} > Select the file Type : </label>
+                <br />
+                {
+                    files.map((file) => {
+                        return (
+                            <button
+                                onClick={() => setFileType(file)}
+                                style={{ backgroundColor: fileType == file ? "#103f5f" : "#6666FF" }}
```
Comment (anushbhatia) [inline src/pages/Editor/sections/Settings/DownloadFileModal.js:58]: use ===

File: src/pages/Editor/sections/Settings/DownloadFileModal.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap';
```
Comment (anushbhatia) [inline src/pages/Editor/sections/Settings/DownloadFileModal.js:2]: here modal, modal body and modal footer aren't used remove them.

**Review by anushbhatia — CHANGES_REQUESTED** [review]
Also, check why the build is failing.

**Comment by nishant23122000** [comment]
Hey @anushbhatia  now it's done, please check it

**Comment by smaranjitghose** [comment]
## Feeback:

- Modify the UI as shown in the image below

Download_UI_Change 

     - The Cancel button can be removed from the bottom and placed on the top with a X button
     - Changing the helper text to "What would you like to call the file?"
     - Make the text above the buttons : "Export as:"

**Comment by smaranjitghose** [comment]
@anushbhatia perhaps you could give some deeper insights into the UI aspect

**Comment by nishant23122000** [comment]
Hey @smaranjitghose I think this close button looks pretty good compare to the close button before and also I change the color of the buttons to the theme color of website

**Comment by smaranjitghose** [comment]
@nishant23122000 LGTM! Could you please pull the recent updates and incorporate them into your PR to fix the merge conflicts. The rest of it is awesome brother!

**Comment by nishant23122000** [comment]
Hey, @smaranjitghose  I think now the conflict is solved and thanks

**Comment by smaranjitghose** [comment]
Nice work @nishant23122000 !

**Comment by nishant23122000** [comment]
@smaranjitghose  Than you😎

**Pull request merged**